### PR TITLE
Development environment updates

### PR DESCRIPTION
**Pull request by replaceafill**
This makes the following changes to the Docker environment:

* Installs Python 3.6 using [`pyenv`](https://github.com/pyenv/pyenv).
* Migrates the old [Compose V1 configuration](https://docs.docker.com/compose/migrate/) and upgrades dependencies.
* Adds arguments for setting the `archivematica` user and group IDs so the content created in the external volumes can be owned by the Docker host user.
* Removes unnecessary prefixes from the `tox` environments.
* Updates the `hack/README.md` file to reflect these updates.

Depends on https://github.com/artefactual/archivematica-storage-service/pull/649
Connected to https://github.com/archivematica/Issues/issues/1612
Connected to https://github.com/archivematica/Issues/issues/1557

### CODE REVIEW

File: hack/docker-compose.yml [hunk 1]
```diff
@@ -16,11 +16,13 @@ volumes:
...
-    external:
-      name: "am-pipeline-data"
+    name: "am-pipeline-data"
+    external: true
+
   archivematica_storage_service_location_data:
-    external:
-      name: "ss-location-data"
+    name: "ss-location-data"
+    external: true
```
Comment (sevein) [inline hack/docker-compose.yml:24]: Is this getting rid of that annoying warning? Awesome!

File: hack/docker-compose.acceptance-tests.yml [hunk 2]
```diff
@@ -1,5 +1,4 @@
 ---
-version: "2.1"
```
Comment (sevein) [inline hack/docker-compose.acceptance-tests.yml:2]: What's the benefit of dropping this attribute?

**Review by sevein — APPROVED** [review]
Lots of nice changes, including pyenv! :partying_face:

File: hack/docker-compose.acceptance-tests.yml (hunk 2 above)
Comment (replaceafill) [reply hack/docker-compose.acceptance-tests.yml:2]: No benefit I guess, it's just that I noticed it's deprecated now:
* https://docs.docker.com/compose/compose-file/04-version-and-name/
* https://github.com/compose-spec/compose-spec/issues/13

Comment (sevein) [reply hack/docker-compose.acceptance-tests.yml:2]: Makes sense.

**Pull request merged**